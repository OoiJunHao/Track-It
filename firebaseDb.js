import * as firebase from 'firebase';
import firestore from 'firebase/firestore';

const firebaseConfig = {
    apiKey: "AIzaSyCLrqVTtM-xrJBDpluCCOjE6_kjEQpibPA",
    authDomain: "track-it-e2a56.firebaseapp.com",
    databaseURL: "https://track-it-e2a56.firebaseio.com",
    projectId: "track-it-e2a56",
    storageBucket: "track-it-e2a56.appspot.com",
    messagingSenderId: "147204621594",
    appId: "1:147204621594:web:3006b19f0054a2223298a3"
};

firebase.initializeApp(firebaseConfig);
firebase.firestore();

export default firebase;