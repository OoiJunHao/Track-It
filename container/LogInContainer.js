import React, { Component } from "react";
import {
  StyleSheet,
  View,
  TextInput,
  Text,
  ScrollView,
  Image,
  KeyboardAvoidingView,
  Keyboard,
  TouchableWithoutFeedback,
  SafeAreaView,
} from "react-native";
import firebaseDb from "../firebaseDb";
import Button from "../component/Button";
import PasswordTextBox from "../component/PasswordTextBox";
import TextBox from "../component/TextBox";
import { Item } from "native-base";

class LogInContainer extends Component {
  state = {
    name: "",
    password: "",
  };

  handleUpdateName = (name) => this.setState({ name });
  handleUpdateEmail = (email) => this.setState({ email });
  handleUpdatePassword = (password) => this.setState({ password });
  handleSignIn = (name, password) =>
    firebaseDb
      .firestore()
      .collection("users")
      .where("name", "==", name)
      .where("password", "==", password)
      .get()
      .then((snapshot) => {
        if (snapshot.empty) {
          console.log("No Such Account");
          alert("No such account");
        } else {
          console.log("Logging In");
          this.setState({
            name: "",
            password: "",
          });
          this.props.navigation.navigate("Home");
        }
      })
      .catch((err) => {
        console.log("Error getting documents", err);
      });

  render() {
    return (
      <SafeAreaView style={{ flex: 1 }}>
        <TouchableWithoutFeedback
          onPress={() => {
            Keyboard.dismiss();
            console.log("dismissed keyboard");
          }}
        >
          <KeyboardAvoidingView behavior="padding" style={styles.container}>
            <ScrollView
              contentContainerStyle={{
                flex: 1,
              }}
            >
              <View style={styles.logoContainer}>
                <Image
                  style={styles.logo}
                  source={require("../assets/logo.png")}
                />
                <View style={styles.inputContainer}>
                  <TextBox
                    icon="person"
                    label="Username"
                    style={(styles.input, { marginBottom: 25, padding: 10 })}
                    onChange={this.handleUpdateName}
                    value={this.state.name}
                    returnKeyType="next"
                    onSubmitEditing={() => this.refs.passwordInput.focus()}
                  />
                  <PasswordTextBox
                    icon="lock"
                    label="Password"
                    onChange={this.handleUpdatePassword}
                    value={this.state.password}
                    style={[styles.input, { marginBottom: 25, padding: 10 }]}
                    ref={"passwordInput"}
                    returnKeyType="go"
                  />
                  <Button
                    style={styles.buttonContainer}
                    onPress={() => {
                      Keyboard.dismiss();
                      if (
                        this.state.name.length &&
                        this.state.password.length
                      ) {
                        this.handleSignIn(this.state.name, this.state.password);
                      } else {
                        alert("Please enter username and password");
                      }
                    }}
                  >
                    <Text>Log In</Text>
                  </Button>
                  <Button
                    onPress={() => {
                      Keyboard.dismiss();
                      this.props.navigation.navigate("Sign Up");
                    }}
                  >
                    <Text>Sign Up</Text>
                  </Button>
                </View>
              </View>
            </ScrollView>
          </KeyboardAvoidingView>
        </TouchableWithoutFeedback>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    backgroundColor: "white",
    alignItems: "center",
    justifyContent: "center",
  },
  logoContainer: {
    alignItems: "center",
    flexGrow: 1,
    justifyContent: "center",
  },
  logo: {
    width: 270,
    height: 110,
    resizeMode: "contain",
  },
  inputContainer: {
    width: 360,
    padding: 20,
    marginTop: 16,
  },
  input: {
    height: 40,
    width: 270,
    backgroundColor: "#d7dedc",
    marginBottom: 8,
    color: "black",
    paddingHorizontal: 10,
    borderColor: "gray",
    borderWidth: 1,
  },
  buttonContainer: {
    backgroundColor: "#2980b9",
    paddingVertical: 16,
    marginBottom: 8,
    marginTop: 8,
  },
  buttonText: {
    textAlign: "center",
    color: "black",
    fontWeight: "700",
  },
});

export default LogInContainer;
