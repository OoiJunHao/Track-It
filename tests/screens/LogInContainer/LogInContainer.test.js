import React from 'react';
import { render, fireEvent, cleanup } from 'react-native-testing-library';
import LogInContainer from '../../../container/LogInContainer';
import renderer from 'react-test-renderer'
import firebaseDb from '../../../firebaseDb';

describe('sample tests', () => {
    it('2 + 2 should be 4', () => {
        expect(2 + 2).toBe(4)
    });
});

afterEach(cleanup);

jest.mock('react-native-firebase');

describe('<LogInContainer />', () => {
    it('should match snapshot', () => {
        const result = render(<LogInContainer />).toJSON();
        expect(result).toMatchSnapshot();
    });

    it('receives email input', () => {
        const rendered = render(<LogInContainer />);
        const textComponent = rendered.getByText('Email');
        expect(textComponent.props.children).toEqual('Email');
    });

    it('should change the password value', () => {
        let loginComponent = renderer.create(<LogInContainer />).getInstance()
        loginComponent.handleUpdatePassword('123456')
        expect(loginComponent.state.password).toEqual('123456')
    })

    it('should change the email value', () => {
        let loginComponent = renderer.create(<LogInContainer />).getInstance()
        loginComponent.handleUpdateEmail('123456')
        expect(loginComponent.state.email).toEqual('123456')
    })

    /* it('should fire onSubmit events', () => {
        const { handleSignIn } = jest.spyOn(handleSignIn)
        const rendered = render(<LogInContainer />)
        const buttonComponent = rendered.getTestID('submit')

        fireEvent(buttonComponent, 'press')

        expect(handleSignIn).toHaveBeenCalled()
    }) */
});

/* describe('handleUpdateEmail function', () => {
    it('should properly render the email input', () => {
        const rendered = render(<LogInContainer />);
        const textComponent = rendered.getByTestId('email');
        expect(textComponent.props.children).toEqual('password');
    });

    it('should fire onChange events', () => {
        const rendered = render(<LogInContainer />);
        const inputComponent = rendered.getByTestId('email');

        fireEvent(inputComponent, 'changeText', 'new text');
        expect(handleUpdateEmail).toHaveBeenCalledWith('new text');
    });
}); */

//verify if user will be able to login with a valid email and valid password

/* describe('User', () => {
    it('is able to login with valid email and password', () => {

        })
    })
}); */





//verify if user cannot login with a valid email and invalid password


//verify the login page for both, when the field is blank and submit button is clicked


//verify messages for invalid login


//verify if data in password field is either vidible as asterisk or bullet signs


//verify if the "enter" key of the keybaord is working correctly on the login page
