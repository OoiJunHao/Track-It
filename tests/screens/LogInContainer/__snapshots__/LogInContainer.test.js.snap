// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogInContainer /> should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView>
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": "100%",
              "position": "absolute",
              "resizeMode": "contain",
              "top": -170,
              "width": "100%",
            }
          }
        />
        <Image
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 110,
              "marginTop": "20%",
              "resizeMode": "contain",
              "width": 270,
            }
          }
        />
        <View
          style={
            Object {
              "marginBottom": 48,
              "marginHorizontal": 30,
            }
          }
        >
          <View
            testID="wrapper"
          >
            <Text
              style={
                Object {
                  "color": "#8A8F9E",
                  "fontSize": 10,
                  "textTransform": "uppercase",
                }
              }
              testID="text"
            >
              Email
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              keyboardType="email-address"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "borderBottomColor": "#8A8F9E",
                  "borderBottomWidth": 0.5,
                  "color": "#161f3d",
                  "fontSize": 15,
                  "height": 40,
                }
              }
              testID="input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "marginBottom": 30,
                "marginTop": 32,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#8A8F9E",
                  "fontSize": 10,
                  "textTransform": "uppercase",
                }
              }
            >
              Password
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomColor": "#8A8F9E",
                  "borderBottomWidth": 0.5,
                  "color": "#161f3d",
                  "fontSize": 15,
                  "height": 40,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#70bee2",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginHorizontal": 30,
                "opacity": 1,
                "paddingVertical": 15,
              }
            }
            testID="submit"
          >
            <Text
              style={
                Object {
                  "color": "rgb(108,108,108)",
                  "fontWeight": "700",
                  "textAlign": "center",
                }
              }
              testID="text"
            >
              <Text>
                Log In
              </Text>
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#70bee2",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginHorizontal": 30,
                "opacity": 1,
                "paddingVertical": 15,
              }
            }
            testID="submit"
          >
            <Text
              style={
                Object {
                  "color": "rgb(108,108,108)",
                  "fontWeight": "700",
                  "textAlign": "center",
                }
              }
              testID="text"
            >
              <Text>
                Sign Up
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
